app/osparc.py

```diff
@@ -0,0 +1,103 @@
+from app.config import Config
+import json
+import osparc
+import tempfile
+from time import sleep
+
+
+def run_simulation(model_url, json_config):
+    temp_config_file = tempfile.NamedTemporaryFile(mode="w+")
+
+    json.dump(json_config, temp_config_file)
+
+    temp_config_file.seek(0)
+
+    try:
+        api_client = osparc.ApiClient(osparc.Configuration(
+            host=Config.OSPARC_API_URL,
+            username=Config.OSPARC_API_KEY,
+            password=Config.OSPARC_API_SECRET
+        ))
+
+        # Upload the configuration file.
+
+        files_api = osparc.FilesApi(api_client)
+
+        try:
+            config_file = files_api.upload_file(temp_config_file.name)
+        except:
+            raise Exception(
+                "the simulation configuration file could not be uploaded")
```

Comment:
You should be raising your own `SimulationException` here and catching that below.  Using `Exception` is the wrong thing to do.

Reply:
Done in https://github.com/agarny/sparc-api/commit/4ce5e7e3fce1ba774dcdcaa8420483a51d65ee1c.

FTR, why is using `Exception` directly the wrong thing to do?

Reply:
It is too broad every exception in Python is an `Exception`.

Reply:
Fair enough.